tests/unit/api/test_upload_sessions.py

```diff
@@ -168,6 +194,52 @@ async def test_list_sessions(
         assert result.data[0].write_mode == UploadSessionWriteModeEnum.KEEP
         assert result.data[0].status == UploadSessionStatusEnum.OPEN
 
+    async def test_list_sessions_with_retry(
```

Comment:
we should also add a test that tests the case where every response returns a 502

Reply:
Added a parametrize